### ISSUE TITLE
💎 Letta 1.0 Stable Release 

### ISSUE THREAD

**Issue by sarahwooders** [letta-ai/letta#1533]
# Letta (MemGPT) 1.0 Stable Release

**Update 01/28/25: We expect only minor modifications to be made to our [existing API](https://docs.letta.com/api-reference/overview), and will be announcing a 1.0 soon. Please see our updated [Python](https://github.com/letta-ai/letta-python) and [Typescript](https://github.com/letta-ai/letta-node) SDKs to use this the latest API.** 

We are working to release a stable version of MemGPT in a 1.0 release, which will be intended for production use. The release will include: 

- Refactoring of DB code (e.g. the `MetadataStore`) to solve scaleability and connection issues (e.g. #1488)
- Consistent API interface for the REST API and the Python SDK, such as matching parameters and return types 
- Introduce stable database schemas and alembic migrations for future releases
- Addition of production-grade testing, logging, and access control

To view progress on the release, you can check the [integration](https://github.com/cpacker/MemGPT/tree/integration) branch. 

### General Cleanup

- Remove all nonessential code, dead or retired code, unofficially supported code from the codebase
- Isolate non-core elements into package add-ons
- Refactor MemGPT configuration to separate out server (e.g. DB) configuration and LLM/embedding configuration

### Logging

- Singular parent logger throughout codebase and removal of all `print` and `printd` calls (except for in the CLI)
- Consistent use of [py logging best practices](https://docs.python.org/3/howto/logging-cookbook.html#adding-handlers-other-than-nullhandler-to-a-logger-in-a-library) for libraries and applications

### Data Schemas

- Consistent data schemas (e.g. `AgentState`) across CLI, Python Client, and REST API represented with pydantic models
- Manage all persisted state (e.g. agent state) in an ORM
- Supported DB migrations for all future versions using alembic

### Auth + Access

- Removal of authentication and access control (determined to be out-of-scope: developers should implement this within their own applications) 

### Testing

- Automated testing for all officially supported integrations (LLM/embedding provider, storage provider) - remove any untested integrations (suggest migrating to adapters or fork)
- SLA Support:
    - Atomic unit and integration tests (should not depend on a live OpenAI API key for testing)
    - Static code analysis threshold [prospector](https://prospector.landscape.io/en/master/) or components of)
- Full CI run for each patch within all supported minors for Python client (versions 3.10, 3.11, 3.12)

### Documentation 
- Add auto-documentation #1493

## Timeline
Planned for end of January 2025: 
- [x]  Database refactor and creation of new schema models 
  - [x] Support auto-migrations with Alembic
- [x]  #1579 
    - [x]  Server
    - [x]  Python Client
    - [x]  REST API
- [x] REST API versioning #1734 
- [x] Autodocs for Python SDK 
- [x]  Refactor configuration
- [x]  Move integrations to external adapters / package add-ons
- [x]  Logging
- [ ] Release 1.0 API

**Comment by a67793581** [comment]
At present, I find that there is a problem with the generation of the open API. It cannot generate the correct parameters. Therefore, I can only manually modify the open API JSON file. The following is the /admin/users interface that I modified. It does not generate the body parameters for the GET request. 
```json

    "/admin/users": {
      "get": {
        "tags": [
          "admin"
        ],
        "summary": "Get All Users",
        "description": "Get a list of all users in the database",
        "operationId": "get_all_users_admin_users_get",
        "security": [
          {
            "HTTPBearer": []
          }
        ],
        "requestBody": {
          "description": "Parameters to filter users.",
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/GetAllUsersRequest"
              }
            }
          }
        },
      "GetAllUsersRequest": {
        "type": "object",
        "properties": {
          "cursor": {
            "type": "string",
            "format": "uuid",
            "description": "Cursor to which to start the paginated request."
          },
          "limit": {
            "type": "integer",
            "description": "Maximum number of users to retrieve per page."
          }
        },
        "required": [],
        "example": {
          "cursor": "00000000-0000-0000-0000-000000000000",
          "limit": 50
        }
      },

**Comment by goetzrobin** [comment]
Outstanding API issues to get dev portal chat UI fully functional:
- [ ] #1639 
- [ ] #1640
- [ ] #1641 
- [ ] #1642 
- [ ] #1643 
- [ ] #1644 
- [ ] #1645 
- [ ] #1646
- [ ] #1647 
- [ ] #1648 
- [ ] #1649

**Comment by Deng-Xian-Sheng** [comment]
Are there any plans to publish something like the OpenAI API so it can be easily integrated into any existing project?

**Comment by sarahwooders** [comment]
@Deng-Xian-Sheng yes we will be releasing documentation on our stable API release soon. You can see a preview here https://docs.letta.com/api-reference/agents/create

**Comment by wuzhongdehua** [comment]
It seems like it’s been a while since the end of January. When will Letta release the stable version 1.0.0? Thanks.
